src/lib/holocene/form/factories/index.ts

```diff
@@ -0,0 +1,20 @@
+import { createServerForm } from './server-form';
+import { createSPAForm } from './spa-form';
+import type { FormConfig, FormContext } from '../types';
+
+export function createFormContext(config: FormConfig): FormContext {
+  switch (config.type) {
+    case 'spa':
+      return createSPAForm(config);
+    case 'server':
+      return createServerForm(config);
+    default: {
+      const _exhaustive: never = config;
```

Comment:
What's up with needing this?

Reply:
its a way of throwing a typescript error if an unhandled type is ever passed to the method. so maybe i add AdminFormConfig to the union but forget to add a factory. then typescript will complain because AdminFormConfig will resolve to never which does not match.